nixos/modules/hardware/device-tree.nix

```diff
@@ -4,6 +4,28 @@ with lib;
 
 let
   cfg = config.hardware.deviceTree;
+  overlayType = types.submodule {
+    options = {
+      overlay = mkOption {
+        type = types.either (types.enum [ "-" ]) types.path;
+        description = ''
+          The overlay file to apply, or '-' to apply parameters to the current
+          device-tree.
+        '';
+        example = literalExample "[\"\${pkgs.linuxPackages_rpi.kernel}/dtbs/overlays/w1-gpio.dtbo\"]";
+      };
```

Comment:
Looks like `linuxPackages_rpi` is no more and you need to say which one e.g. `linuxPackages_rpi4`